### DISCUSSION TITLE
How to connect to aws elastic cache cluster ?

### DISCUSSION THREAD

**Discussion by SeyyedKhandon** [NodeRedis/node_redis#2768] [Q&A] [answered]
I wonder what is the equivalent of  https://github.com/redis/ioredis?tab=readme-ov-file#special-note-aws-elasticache-clusters-with-tls in node-redis?

Currently in ioredis we just use and it works fine( the only issue is https://github.com/redis/ioredis/issues/1454):

```typescript
  const redisCluster = new Redis.Cluster([{ 
host:"clustercfg.redis-cluster-name.xiko.euw1.cache.amazonaws.com:6379",
 port:6379 
}], {
    scaleReads: 'all',
    dnsLookup: (address, callback) => callback(null, address),
    redisOptions: {
      password:"token",
      tls: {},
    },
  })

```

**Comment by leibale** [comment, marked as answer]
I'm not too sure why the `dnsLookup` function is needed, but using `RedisClient`, anything that is under the `socket` option is passed directly to `net.connect` or `tls.connect` (see https://github.com/redis/node-redis/blob/master/docs/client-configuration.md), so you can do:
```javascript
createClient({
  socket: {
    host: 'redis.server',
    port: 6379
    tls: true
  },
  password: 'token'
});
```
and using `RedisCluster`, anything under `defaults` will be passed to all clients created by the cluster (see https://github.com/redis/node-redis/blob/master/docs/clustering.md#createcluster-configuration), so you can do:
```javascript
createCluster({
  rootNodes: [{
    socket: {
      host: 'redis.server',
      port: 6379
    }
  }],
  defaults: {
    socket: {
      tls: true
    },
    password: 'token'
  }
});
```

please let me know if it works for you, and if and what you think we should update in the documentation

edit:
I thought `dnsLookup` is an option ioredis passes to `net.connect/tls.connect` but that's not true.. As I've said, I'm not too sure if its needed, I suggest you try it without it.

**Reply by SeyyedKhandon** [reply]
Well using dns is based on the `ioredis` doc and it is necessary for aws elastic cache.

And for createCluster from node-redis, actually there is no `host, port` options available on node-redis, and it is only url or either socket

**Reply by leibale** [reply]
I've edited my answer to show how to do `host` and `port`

**Reply by SeyyedKhandon** [reply]
Thanks, it worked.

**Reply by SeyyedKhandon** [reply]
@leibale Could you please have a look into this pr https://github.com/redis/node-redis/pull/2771 which is about adding this example of yours into the doc.
